msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/DiscussionForumServiceImpl.java

```diff
@@ -602,362 +665,444 @@ public String getLabel()
+					+ message.getId(), e);
+			throw e;
+		}
+		message.setUuid(messageElement.getAttribute(MESSAGE_UUID));
+		message.setDeleted(nullSafeToBoolean(messageElement.getAttribute(MESSAGE_DELETED)));
+		message.setHasAttachments(nullSafeToBoolean(messageElement.getAttribute(MESSAGE_HAS_ATTACHMENTS)));
+		message.setApproved(nullSafeToBoolean(messageElement.getAttribute(MESSAGE_APPROVED)));
+		message.setGradeAssignmentName(messageElement.getAttribute(MESSAGE_GRADE_ASSIGNMENT_NAME));
+		message.setLabel(messageElement.getAttribute(MESSAGE_LABEL));
+		message.setNumReaders(nullSafeToInteger(messageElement.getAttribute(MESSAGE_NUM_READER)));
+		message.setThreadId(nullSafeToLong(messageElement.getAttribute(MESSAGE_THREAD_ID)));
+		message.setThreadLastPost(nullSafeToLong(messageElement.getAttribute(MESSAGE_THREAD_LAST_POST)));
+		final String dateThreadLastUpdated = messageElement.getAttribute(MESSAGE_DATE_THREAD_LAST_UPDATED);
+		if (StringUtils.isNotBlank(dateThreadLastUpdated)) {
+			try {
+				message.setDateThreadlastUpdated(formatter.parse(dateThreadLastUpdated));
+			} catch (ParseException e) {
+				log.error(
+						"ERROR merging messages: Wrong date format or null in thread last updated in message with id: "
+								+ message.getId());
+				throw e;
+			}
+		}
+		if (StringUtils.isNotBlank(messageIdInReplyTo)) {
+			message.setInReplyTo(messageManager.getMessageById(Long.valueOf(messageIdInReplyTo)));
+		}
+		message.setBody(getDecodedString(messageElement.getAttribute(MESSAGE_BODY)));
+
+		// Set the topic of the message
+		message.setTopic(discussionTopic);
+		return message;
+	}
+
+	private String getDescriptionFromPropertyElement(final Element propertyElement) {
+		if (BASE64.equals(propertyElement.getAttribute(ENCODE))) {
+			return getDecodedString(propertyElement.getAttribute(VALUE));
+		} else {
+			return propertyElement.getAttribute(VALUE);
+		}
+	}
+
+	private String getDecodedString(final String inputString) {
+		if (StringUtils.isNotBlank(inputString)) {
+			return new String(base64Encoder.decode(inputString.getBytes()), StandardCharsets.UTF_8);
+		}
+		return StringUtils.EMPTY;
+	}
+
+	private String getEncodedString(final String inputString) {
+		try {
+			return new String(base64Encoder.encode(inputString.getBytes()));
+		} catch (Exception e) {
+			return StringUtils.EMPTY;
 		}
-		return null;
+	}
+
+	private Long nullSafeToLong(String stringToConvert) {
+		Long longToReturn = null;
+		if (StringUtils.isNotBlank(stringToConvert)) {
+			longToReturn = Long.valueOf(stringToConvert);
+		}
+		return longToReturn;
+	}
+
+	private Boolean nullSafeToBoolean(String stringToConvert) {
+		Boolean booleanToReturn = null;
+		if (StringUtils.isNotBlank(stringToConvert)) {
+			booleanToReturn = Boolean.valueOf(stringToConvert);
+		}
+		return booleanToReturn;
+	}
+
+	private Integer nullSafeToInteger(String stringToConvert) {
+		Integer integerToReturn = null;
+		if (StringUtils.isNotBlank(stringToConvert)) {
```

Comment:
Maybe use StringUtils.isNumeric which makes it a little more safer ;)

Reply:
Yes! Thanks for the tip!